src/node_crypto.cc

```diff
@@ -988,19 +988,31 @@ static X509_STORE* NewRootCertStore() {
 
 void GetRootCertificates(const FunctionCallbackInfo<Value>& args) {
   Environment* env = Environment::GetCurrent(args);
-  Local<Value> result[arraysize(root_certs)];
 
-  for (size_t i = 0; i < arraysize(root_certs); i++) {
-    if (!String::NewFromOneByte(
-            env->isolate(),
-            reinterpret_cast<const uint8_t*>(root_certs[i]),
-            NewStringType::kNormal).ToLocal(&result[i])) {
-      return;
+  if (root_cert_store == nullptr)
+    root_cert_store = NewRootCertStore();
+
+  stack_st_X509_OBJECT* objs = X509_STORE_get0_objects(root_cert_store);
+  int num_objs = sk_X509_OBJECT_num(objs);
+
+  std::vector<Local<Value>> result;
+  result.reserve(num_objs);
+
+  for (int i = 0; i < num_objs; i++) {
+    X509_OBJECT* obj = sk_X509_OBJECT_value(objs, i);
+    if (X509_OBJECT_get_type(obj) == X509_LU_X509) {
+      X509* cert = X509_OBJECT_get0_X509(obj);
+
+      Local<Value> value;
+      if (!X509ToPEM(env, cert).ToLocal(&value))
+        return;
```

Comment:
Might be worthwhile including  `CHECK` in here to verify that an exception is pending.

Reply:
What's the best way for doing that? 

Digging around a bit I could find `IsExceptionPending` in NAPI and a V8 `has_pending_exception()` on `Isolate`. Can't find any existing use of the V8 function anywhere in the Node.js source.

Reply:
`CHECK(env->isolate()->has_pending_exception()` should work. And yeah, it's not common because we've been really bad historically about handling this bit ;-) ... I think we're trying to do better at it and we've got to start somewhere 😄

Reply:
Turns out `has_pending_exception()` only exists on the internal variant of `isolate`.